### ISSUE TITLE
Support new DDB Table PITR Property RecoveryPeriodInDays

### ISSUE THREAD

**Issue by scolebrook** [aws-cloudformation/cfn-lint#3902]
### CloudFormation Lint Version

1.22.3

### What operating system are you using?

macOS

### Describe the bug

Templates with a valid DynamoDB Table with the new `PointInTimeRecoverySpecification.RecoveryPeriodInDays` property fails validation.

```yaml
      PointInTimeRecoverySpecification:
        PointInTimeRecoveryEnabled: true
        RecoveryPeriodInDays: 7
```

Error message
`E3002 Additional properties are not allowed ('RecoveryPeriodInDays' was unexpected)`

### Expected behavior

Successful validation

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'

Resources:
  table:
    Type: "AWS::DynamoDB::Table"
    Properties:
      TableName: "MyTable"
      AttributeDefinitions:
        - AttributeName: "pk"
          AttributeType: "S"
      KeySchema:
        - AttributeName: "pk"
          KeyType: "HASH"
      PointInTimeRecoverySpecification:
        PointInTimeRecoveryEnabled: true
        RecoveryPeriodInDays: 7
```

**Comment by scolebrook** [comment]
Fixed in 1.22.4
